client_test.go

```diff
@@ -0,0 +1,83 @@
+package client
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+
+func TestNewClient(t *testing.T) {
+	Convey("Given acceptable input values", t, func(){
+		c, err := NewClient("https://localhost:4646", "testdata/testCertFile", false)
+		Convey("When a newClient is created", func() {
+			Convey("Then a client is returned with no errors", func(){
+				So(c, ShouldNotBeNil)
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+}
+
+func TestNewClientFails(t *testing.T) {
+	Convey("Given invalid input values", t, func(){
```

Comment:
`"Given no nomadCACert is provided and nomadTLSSkipVerify is false"` is more descriptive and informs the reader specifically what scenario this test is validating.